fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1376,12 +1391,268 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
 
         whereClauseSegment.append("(SELECT 1 FROM " + valuesTable + WHERE);
         this.populateNameIdSubSegment(whereClauseSegment, queryParm.getCode(), valuesTable.toString());
-        whereClauseSegment.append(" AND " + valuesTable + ".LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID"); // correlate the [NOT] EXISTS subquery
+        whereClauseSegment.append(AND).append(valuesTable).append(".LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID"); // correlate the [NOT] EXISTS subquery
         whereClauseSegment.append(RIGHT_PAREN).append(RIGHT_PAREN);
 
         List<Object> bindVariables = new ArrayList<>();
         SqlQueryData queryData = new SqlQueryData(whereClauseSegment.toString(), bindVariables);
         log.exiting(CLASSNAME, METHODNAME);
         return queryData;
     }
+
+    /**
+     * Contains special logic for handling reverse chained reference search parameters.
+     * <p>
+     * A select statement is built to realize the reverse chaining logic required. Here is a sample
+     * reverse chained query for a Patient given this search parameter: _has:Observation:patient:code=1234
+     *
+     * <pre>
+     * SELECT
+     *   CLR0.LOGICAL_ID
+     * FROM
+     *   Patient_LOGICAL_RESOURCES AS CLR0
+     *   JOIN Patient_RESOURCES AS CR0 ON CR0.RESOURCE_ID = CLR0.CURRENT_RESOURCE_ID AND CR0.IS_DELETED = 'N'
+     * WHERE
+     *   EXISTS (
+     *     SELECT
+     *       1
+     *     FROM
+     *       Observation_TOKEN_VALUES_V AS CP1
+     *       JOIN Observation_LOGICAL_RESOURCES AS CLR1 ON CLR1.LOGICAL_RESOURCE_ID = CP1.LOGICAL_RESOURCE_ID
+     *       JOIN Observation_RESOURCES AS CR1 ON CR1.RESOURCE_ID = CLR1.CURRENT_RESOURCE_ID AND CR1.IS_DELETED = 'N'
+     *       JOIN Observation_TOKEN_VALUES_V AS CP2 ON CP2.LOGICAL_RESOURCE_ID = CLR1.LOGICAL_RESOURCE_ID
+     *       AND (
```

Comment:
I don't think we should use parens unless required to control precedence.

Reply:
Agreed. Unfortunately in this case they're generated by the populateNameIdSubSegment() method, which gets called by many other methods - didn't want to take on making a change there and having to validate all of those existing cases.